### PR TITLE
Implement api.json generation in pulp-docs

### PR DESCRIPTION
**Pull request by pedro-psb**
About this PR:
- Implement a module to generate all or a subset of `{plugin}-api.json` using `pulpcore-manager openapi` command.
- Add basic test to this module
- Add a workflow to run the pytest suite on CI

### CODE REVIEW

**Review by dkliban — APPROVED** [review]

File: src/pulp_docs/openapi.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        install_cmd = []
+        create_venv_cmd = ("python", "-m", "venv", self.venv_path)
+
+        if plugin.is_subpackage or plugin.name == "pulpcore":
+            if Path(self.venv_path).exists():
+                return
+            install_cmd = ["pip", "install", "pulpcore"]
+        else:
+            url = f"git+{plugin.get_remote_url()}"
+            install_cmd = ["pip", "install", "pulpcore", url]
```
Comment (mdellweg) [inline src/pulp_docs/openapi.py:105]: Each plugin depends on pulpcore. Why adding that here?

File: src/pulp_docs/openapi.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        """
+        Creates virtualenv with plugin + pulpcore.
+        """
+        install_cmd = []
+        create_venv_cmd = ("python", "-m", "venv", self.venv_path)
+
+        if plugin.is_subpackage or plugin.name == "pulpcore":
+            if Path(self.venv_path).exists():
+                return
+            install_cmd = ["pip", "install", "pulpcore"]
```
Comment (mdellweg) [inline src/pulp_docs/openapi.py:102]: We install the others from git why is it different for pulpcore?

File: src/pulp_docs/openapi.py (hunk 1 above)
Comment (pedro-psb) [reply src/pulp_docs/openapi.py:105]: Somehow I've ignored that.

File: src/pulp_docs/openapi.py (hunk 2 above)
Comment (pedro-psb) [reply src/pulp_docs/openapi.py:102]: It shouldnt be.

**Comment by pedro-psb** [comment]
I'll send another PR with those fixes

**Pull request merged**